src/components/Todo.vue

```diff
@@ -0,0 +1,150 @@
+<template>
+  <div class="todo">
+    <template class="todo-intarface">
+      <h1
+        :class="{
+          red: checkCountOfDoneTask() < 0.5,
+          yellow: checkCountOfDoneTask() >= 0.5 && checkCountOfDoneTask() < 1,
+          green: checkCountOfDoneTask() === 1,
+        }"
+      >
+        TODO LIST
+      </h1>
+      <input
+        class="todo-input"
+        v-on:keydown.enter="addTodo"
+        type="text"
+        v-model="todo"
+      />
+      <button class="todo-btn" @click="addTodo">add new task</button>
+    </template>
+    <ul class="todo-list">
+      <li
+        v-for="(t, idx) of todos"
+        :key="idx"
+        class="todo-item"
+        @click.self="removeTodo(idx)"
+      >
+        <input
+          class="todo-checkbox"
+          @change.self="changeStatus(t)"
+          type="checkbox"
+          name="is-done"
+        />
+        <span :class="{ done: t.status }" @click="removeTodo(idx)"
```

Comment:
There is a problem with using indexes instead of unique iterable ids, it becomes apparent when interacting with the list elements, like removing them by id:

1. Add three tasks:
![image](https://user-images.githubusercontent.com/42524280/185459169-bb567ddb-6916-4996-8b95-9ad180eddc12.png)

2. Check second and thid task:
![image](https://user-images.githubusercontent.com/42524280/185459261-833e27ce-c5e3-48de-8646-40415ce95795.png)

3. Remove the first task. See how the checkbox is inverted? It happens because Vue is optimizing this list behind the scenes, and not re-calculating everything.
![image](https://user-images.githubusercontent.com/42524280/185459437-01484e18-f67e-46b9-9ba8-0ad6b25da133.png)

___

The two known ways of fixing this bug are:
- Force the list to re-calculate by adding a new dependency to it.
- Use unique id keys instead of indexes. **👈 This is preferrable and what is usually used in real life applications.**

If you're interested, here's an explanation of what happens here: https://youtu.be/S2wj8VX9O0M